### PR TITLE
Added Go subcommand 'dcos node log'.

### PR DESCRIPTION
**Pull request by armandgrillet**

### CODE REVIEW

File: pkg/logs/client.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"io/ioutil"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/r3labs/sse"
+)
+
+// Client is a metrics client for DC/OS.
```
Comment (rgo3) [inline pkg/logs/client.go]:
```suggestion
// Client is a logs client for DC/OS.
```

File: pkg/logs/client.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+type Client struct {
+	http *httpclient.Client
+}
+
+// NewClient creates a new metrics client.
```
Comment (rgo3) [inline pkg/logs/client.go]:
```suggestion
// NewClient creates a new logs client.
```

**Review by rgo3 — COMMENTED** [review]
Otherwise LGTM

File: pkg/logs/client.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+func (c *Client) PrintComponent(route string, service string, skip int, filters string, follow bool) error {
+	endpoint := fmt.Sprintf("/system/v1%s/logs/v2/component%s?skip=%d%s", route, service, skip, filters)
+	if follow {
+		client := sse.NewClient(c.http.BaseURL().String() + endpoint)
+		client.Connection = c.http.BaseClient()
+		client.Headers["Authorization"] = c.http.Header().Get("Authorization")
+		client.Headers["User-Agent"] = c.http.Header().Get("User-Agent")
+
+		var data SSEEventDataField
+		client.Subscribe("", func(msg *sse.Event) {
```
Comment (bamarni) [inline pkg/logs/client.go]:
```suggestion
		return client.SubscribeRaw(func(msg *sse.Event) {
```

File: pkg/logs/client.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+			json.Unmarshal(msg.Data, &data)
+			date := time.Unix(data.RealtimeTimestamp/int64(math.Pow(10, 6)), 0)
+			fmt.Println(date.Format("2006-01-02 15:04:05 MST") + ": " + data.Fields.Message)
```
Comment (bamarni) [inline pkg/logs/client.go]: Can you make it configurable that it doesn't always prints to stdout? To allow unit testing.

File: pkg/logs/client.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == 200 {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		fmt.Println(strings.TrimSpace(string(bodyBytes)))
```
Comment (bamarni) [inline pkg/logs/client.go]: same here, `fmt.Fprintln(c.out, strings.TrimSpace(string(bodyBytes)))`

File: pkg/cmd/node/node_log.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+				route = "/agent/" + mesosID
+			}
+			service := ""
+			if component != "" {
+				service = fmt.Sprintf("/%s.service", component)
+			}
+			requestFilters := ""
+			if len(filters) > 0 {
+				requestFilters = "&filter=" + strings.Join(filters, "&filter=")
+			}
```
Comment (bamarni) [inline pkg/cmd/node/node_log.go]: I think it'd be better to pass the filters as is (as a slice) to the client function, and the client would be in charge of constructing the URL with the query parameters.

**Comment by bamarni** [comment]
It'd be nice to have some test coverage too, we don't have a single unit test on the Go code a the moment...

**Comment by bamarni** [comment]
My bad, we have one actually https://github.com/dcos/dcos-core-cli/blob/1.13-patch.x/pkg/mesos/types_test.go#L9 :tada:

File: pkg/logs/client.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	if follow {
+		client := sse.NewClient(c.http.BaseURL().String() + endpoint)
+		client.Connection = c.http.BaseClient()
+		client.Headers["Authorization"] = c.http.Header().Get("Authorization")
+		client.Headers["User-Agent"] = c.http.Header().Get("User-Agent")
+
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, os.Interrupt)
+		events := make(chan *sse.Event)
+		client.SubscribeChanRaw(events)
```
Comment (bamarni) [inline pkg/logs/client.go]: this can return an error

File: pkg/logs/client.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+		var data SSEEventDataField
+
+		for {
+			select {
+			case msg := <-events:
+				json.Unmarshal(msg.Data, &data)
```
Comment (bamarni) [inline pkg/logs/client.go]: shall we handle the error here?

**Review by rgo3 — APPROVED** [review]

**Comment by armandgrillet** [comment]
@mesosphere-ci run integration tests

**Review by bamarni — APPROVED** [review]

**Pull request merged**